src/Model/UnvalidatedOidcTokens.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Drenso\OidcBundle\Model;
+
+use DateTimeImmutable;
+use Drenso\OidcBundle\Enum\OidcTokenType;
+use Drenso\OidcBundle\Exception\OidcException;
+use stdClass;
+
+/**
+ * Contains the access and id tokens retrieved from OpenID authentication.
+ */
+class UnvalidatedOidcTokens
+{
+  protected readonly ?string $accessToken;
+  protected readonly ?string $idToken;
+  private ?DateTimeImmutable $expiry = null;
+  private ?string $refreshToken      = null;
+  /** @var string[]|null */
+  private ?array $scope = null;
+
+  /**
+   * @param stdClass|UnvalidatedOidcTokens $tokens
+   * @throws OidcException
+   */
+  public function __construct(object $tokens)
+  {
+    if ($tokens instanceof stdClass) {
+      $this->accessToken = $tokens->access_token ?? null;
+      $this->idToken     = $tokens->id_token ?? null;
+
+      if (isset($tokens->expires_in)) {
+        $this->expiry = DateTimeImmutable::createFromFormat('U', (string)(time() + $tokens->expires_in));
+      }
+
+      if (isset($tokens->refresh_token)) {
+        $this->refreshToken = $tokens->refresh_token;
+      }
+
+      if (isset($tokens->scope)) {
+        $this->scope = explode(' ', (string)$tokens->scope);
+      }
+
+    } elseif ($tokens instanceof UnvalidatedOidcTokens) {
+      $this->accessToken = $tokens->getAccessToken();
+      $this->idToken = $tokens->getAccessToken();
```

Comment:
```suggestion
      $this->accessToken = $tokens->accessToken;
      $this->idToken = $tokens->idToken;
```

And think this holds for the other assignments here as well (property access instead of method access).
```

Reply:
PHP is always good for a surprise. Even after so many years of working with PHP you always discover something new. `private` and `protected` things from another instance are accessible in the same class.

Changed the code as suggested.